### PR TITLE
Cleanup Jenkins-related docs in prow plus a fix

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/6996 plus cleans up the comments a bit.

/cc @stevekuznetsov

### CODE REVIEW

File: prow/jenkins/jenkins.go [hunk 1]
```diff
@@ -112,39 +113,41 @@ func (jb *JenkinsBuild) IsEnqueued() bool {
 }
 
 // ProwJobID extracts the ProwJob identifier for the
-// Jenkins build. We prefer the PROW_JOB_ID parameter
-// if present but fall back to legacy parameters if
-// not.
+// Jenkins build in order to correlate the build with
+// a ProwJob. If the build has an empty PROW_JOB_ID
+// it didn't start by prow.
 func (jb *JenkinsBuild) ProwJobID() string {
```
Comment (stevekuznetsov) [inline prow/jenkins/jenkins.go]: This is breaking and needs a README update

Comment (kargakis) [reply prow/jenkins/jenkins.go]: It's not breaking.

> Furthermore, the $BUILD_ID variable in Jenkins jobs will now correctly point to the build identifier vended by tot and a new variable, $PROW_JOB_ID, points to the identifier used to link ProwJobs to Jenkins builds.

I think it just fixes the above quote to be 100% true.

Comment (kargakis) [reply prow/jenkins/jenkins.go]: In the case PROW_JOB_ID exists in Jenkins, things work, but then you may have other problems
/shrug

Comment (stevekuznetsov) [reply prow/jenkins/jenkins.go]: Hmm yes this is the assumption that I did not consider -- `"Assuming Jenkins is not updated to have the new PROW_JOB_ID variable"` -- if that does not hold, the fallback works.

Comment (kargakis) [reply prow/jenkins/jenkins.go]: Added announcement.

**Pull request merged**